sdk/keyvault/keyvault-certificates/test/identifier.test.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { KeyVaultCertificatesIdentifier } from "../src";
+
+describe("KeyVault Certificates Identifier", () => {
+  it("It should work with a URI of a certificate before it gets a version", async function() {
+    const uri = "https://keyvault-name.vault.azure.net/certificates/certificate-name/pending";
+    const identifier = new KeyVaultCertificatesIdentifier(uri);
+
+    // Assert doesn't have deepEqual on the browser.
+    assert.equal(
+      JSON.stringify(identifier),
+      JSON.stringify({
+        collection: "certificates",
+        id: "https://keyvault-name.vault.azure.net/certificates/certificate-name/pending",
+        vaultUrl: "https://keyvault-name.vault.azure.net",
+        version: "pending",
+        name: "certificate-name"
+      })
+    );
+  });
+
+  it("It should work with a URI of a certificate with a specific version", async function() {
```

Comment:
What about deleted certificate tests?

Reply:
I don't think there's any difference between the URI of a key/secret/certificate and the URI of a deleted key/secret/certificate. I'll confirm though.

Reply:
Here's an ID of a certificate I just deleted and retrieved: `https://danrodri-kv-2020-1.vault.azure.net/certificates/recoverCertificateName-cangetadeletedcertificate-15203725962992753/26d7cc3645b146fe80d7bb56e7aba0d6` (the uniqueness of it has been randomly generated). It's the exact same format of a certificates URL, as far as I can tell.

Reply:
I see where I went wrong: I was thinking of the `recoveryId` once an artifact is deleted (was thinking it became the `id` after deletion). Sorry about that.

Still, if people have a `recoveryId` they may want to parse it to get, for example, the `vaultUrl` and `name` that we require. That said, since `recoveryId` is exposed as a distinct property, maybe it's not worth it.

We do have a tracking issue open for .NET to use our shared parser in more places where we will need this, but we go do that internally only.

Reply:
Oh that's really nice to know! Thank you. I will see how this behaves with the recoveryId.